proving.go

```diff
@@ -219,9 +219,14 @@ func subtreeDefinition(c CacheReader, p Position) (root, firstLeaf Position, wid
 		return Position{}, Position{}, 0, err
 	}
 	maxRootHeight := RootHeightFromWidth(width)
-	for root = p.parent(); root.Height < maxRootHeight; root = root.parent() {
-		if layer := c.GetLayerReader(root.Height); layer != nil {
-			break
+	root = p
+	if !(p.Height == 0 && width == 1) {
+		// TODO(dshulyak) doublecheck with @noam if there is more generic fix
```

Comment:
is this still a TODO or have you done it?

Reply:
still open. but it is a minor issue. i will open a github issue to track it so that either @moshababo / @noamnelke can take a look into this when they have time.